cds/modules/ffmpeg/ffmpeg.py

```diff
@@ -59,14 +66,47 @@ def ff_probe_all(input_filename):
     * *-show_format -print_format json* output in JSON format
     * *-show_streams -select_streams v:0* show information for video streams
     """
-    return check_output([
+    metadata = check_output([
         'ffprobe', '-v', 'error',
         '-show_format', '-print_format', 'json',
         '-show_streams', '-select_streams', 'v:0',
         '{}'.format(input_filename)
     ]).decode('utf-8')
 
+    # Patch invalid aspect ratio
+    info = json.loads(metadata)
+    sinfo = info['streams'][0]
+    key = 'display_aspect_ratio'
+    if sinfo[key] == '0:1':
+        sinfo[key] = calculate_aspect_ratio(sinfo['width'], sinfo['height'])
+    return json.dumps(info)
+
+
+#
+# Aspect Ratio
+#
+def valid_aspect_ratios():
+    return [(16, 9), (4, 3), (3, 2), (20, 9), (256, 135), (64, 35), (2, 1)]
 
+
+def probe_aspect_ratio(input_filename):
+    metadata = ff_probe_all(input_filename)
+    return json.loads(metadata)['streams'][0]['display_aspect_ratio']
+
+
+def calculate_aspect_ratio(width, height):
```

Comment:
Probably this shouldn't be part of the API functions (If we can call then that), what about `_calculate_aspect_ratio` to make it more clear?